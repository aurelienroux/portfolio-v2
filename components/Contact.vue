<template>
  <div class="contact">
    <h2 class="contact__title">Say hi!</h2>
    <div class="contact__links">
      <a href="mailto:aureroux13@gmail.com" class="contact__link">
        <font-awesome-icon icon="envelope" size="6x" />
      </a>
      <a
        href="https://www.linkedin.com/in/rouxaurelien/"
        target="_blank"
        rel="noopener noreferrer"
        class="contact__link"
      >
        <font-awesome-icon :icon="['fab', 'linkedin']" size="6x" />
      </a>
      <a
        href="https://github.com/aurelienroux"
        target="_blank"
        rel="noopener noreferrer"
        class="contact__link"
      >
        <font-awesome-icon :icon="['fab', 'github-alt']" size="6x" />
      </a>
    </div>
  </div>
</template>

<script></script>

<style lang="scss" scoped>
.contact {
  display: flex;
  flex-direction: column;
  justify-content: center;
  min-height: 100vh;

  &__title {
    color: $color-orange;
    font-family: $font-montserrat;
    font-size: 3rem;
    margin: 0 0 6rem 0;
    text-align: center;

    @include for-tablet-portrait-up {
      font-size: 4rem;
      margin: 0 0 14rem 0;
    }
  }

  &__links {
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-content: space-around;

    @include for-tablet-portrait-up {
      flex-direction: row;
    }
  }

  &__link {
    color: $color-text;
    margin: 2rem;
    padding: 2rem;
    transition: 0.3s;

    &:hover {
      color: $color-orange;
      transition: 0.3s;
    }
  }
}
</style>
