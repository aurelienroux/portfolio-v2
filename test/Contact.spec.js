// import { shallowMount } from '@vue/test-utils'
import { shallowMount } from '@vue/test-utils'
import Contact from '@/components/Contact.vue'

const wrapper = shallowMount(Contact, {
  stubs: ['font-awesome-icon'],
})

describe('Contact', () => {
  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  test('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('.contact').exists()).toBe(true)
    expect(wrapper.find('.contact__title').exists()).toBe(true)
  })

  test('should display contact links', () => {
    expect(wrapper.find('.contact__links').exists()).toBe(true)
    expect(wrapper.findAll('.contact__link').length).toBe(4)
  })

  test('should display correct links icons', () => {
    expect(wrapper.findAll('font-awesome-icon-stub').length).toBe(4)
    expect(
      wrapper.findAll('font-awesome-icon-stub').at(0).attributes('icon')
    ).toBe('envelope')
    expect(
      wrapper.findAll('font-awesome-icon-stub').at(1).attributes('icon')
    ).toBe('fab,linkedin')
    expect(
      wrapper.findAll('font-awesome-icon-stub').at(2).attributes('icon')
    ).toBe('fab,github-alt')
    expect(
      wrapper.findAll('font-awesome-icon-stub').at(3).attributes('icon')
    ).toBe('fas,file-pdf')
  })
})
