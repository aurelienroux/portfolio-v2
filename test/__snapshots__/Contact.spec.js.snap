// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact renders correctly 1`] = `
"<div class=\\"contact\\">
  <h2 class=\\"contact__title\\">Say hi!</h2>
  <div class=\\"contact__links\\"><a href=\\"mailto:aureroux13@gmail.com\\" class=\\"contact__link\\">
      <font-awesome-icon-stub icon=\\"envelope\\" size=\\"6x\\"></font-awesome-icon-stub>
    </a> <a href=\\"https://www.linkedin.com/in/rouxaurelien/\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" class=\\"contact__link\\">
      <font-awesome-icon-stub icon=\\"fab,linkedin\\" size=\\"6x\\"></font-awesome-icon-stub>
    </a> <a href=\\"https://github.com/aurelienroux\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" class=\\"contact__link\\">
      <font-awesome-icon-stub icon=\\"fab,github-alt\\" size=\\"6x\\"></font-awesome-icon-stub>
    </a> <a href=\\"/resume.pdf\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" class=\\"contact__link\\">
      <font-awesome-icon-stub icon=\\"fas,file-pdf\\" size=\\"6x\\"></font-awesome-icon-stub>
    </a></div>
</div>"
`;
